packages/service-content/src/lib/blogs/import/main.ts

```diff
@@ -56,22 +56,26 @@ export const createProcessMainFile = (transaction: TransactionSql) => {
     if (!result) {
       throw new Error('Could not insert blog');
     }
-
+    const blogId = result.id;
     if (parsedBlog.tags && parsedBlog.tags.length > 0) {
       await transaction`
-          INSERT INTO content.tags ${transaction(
-            parsedBlog.tags.map((tag) => ({ name: tag.toLowerCase() })),
-          )}
-          ON CONFLICT (name) DO NOTHING
-        `;
+    DELETE FROM content.blog_tags WHERE blog_id = ${blogId}
+  `;
+
+      await transaction`
+    INSERT INTO content.tags ${transaction(
+      parsedBlog.tags.map((tag) => ({ name: tag.toLowerCase() })),
+    )}
+    ON CONFLICT (name) DO NOTHING
+  `;
 
       await transaction`
-          INSERT INTO content.blog_tags (blog_id, tag_id)
-          SELECT
-            ${result.id},
-            id FROM content.tags WHERE name = ANY(${parsedBlog.tags.map((tag) => tag.toLowerCase())})
-          ON CONFLICT DO NOTHING
-        `;
+    INSERT INTO content.blog_tags (blog_id, tag_id)
+    SELECT ${blogId}, id
```

Comment:
You changed indentation on all three queries.
Best is like before: have one more tab than the above `await transaction`
Plus add one tab on sub parts of the query (what is selected, what is inserted, what values, ...).